data/transactions/logic/eval.go

```diff
@@ -4060,3 +4062,126 @@ func opBase64Decode(cx *EvalContext) {
 	}
 	cx.stack[last].Bytes, cx.err = base64Decode(cx.stack[last].Bytes, encoding)
 }
+func hasDuplicateKeys(dec *json.Decoder) (bool, error) {
```

Comment:
Suggestion: make this function take a byte slice instead, since you can't reuse the same decoder after this. Also, it will make unit tests easier.